Dockerfile

```diff
@@ -45,9 +45,6 @@ ADD . /code/
 ENV PORT 8000
 EXPOSE 8000
 
-# Add custom environment variables needed by Django or your settings file here:
-ENV DJANGO_SETTINGS_MODULE=bakerydemo.settings.production DJANGO_DEBUG=off
```

Comment:
Question: Why are we removing the default settings?

Reply:
I think we should still keep the production default in the `Dockerfile`.

Reply:
If there are undocumented use cases that require production-like settings and packages, could that be addressed by someone else who is aware of these scenarios in a separate PR? I can only work from the documented use cases in the readme, which has this to say about the docker compose setup:

https://github.com/wagtail/bakerydemo/blob/cd0910420313e0f18c6b9c2a048b227f1f8b3296/readme.md?plain=1#L112

The readme.md makes it sound like all paths for starting the demo are equivalent and only a matter of preference, but per my previous [comment](https://github.com/wagtail/bakerydemo/pull/464#issuecomment-1808722918):

- Gitpod installs requirements.txt which is essentially just an alias for requirements/production.txt
- Vagrant installs requirements/base.txt
- Docker Compose installs requirements/production.txt
- Virtualenv installs requirements/development.txt

Either the readme needs some reworking or the configurations should be consistent with each other and with the readme. The simplest starting point for demo purposes is to install `base.txt` (and perhaps an extra file for docker to install the redis/psycopg3 libs). 

For anyone who wishes to submit a PR to `bakerydemo`, they'll need to install `development.txt` (removing the `-r base.txt` from that file) in a virtualenv on their _host_ so that they can run `ruff` etc. on their changes before submitting a PR (the contributing docs needs a note about installing `pre-commit`).